nixos/modules/services/cluster/k3s/default.nix

```
@@ -114,6 +115,26 @@ in
       default = null;
       description = "File path containing the k3s YAML config. This is useful when the config is generated (for example on boot).";
     };
+
+    registries = mkOption {
+      type = types.nullOr yaml.type;
+      default = null;
+      example = {
+        mirrors."docker.io".endpoint = ["https://registry.example.com:5000"];
+        configs."registry.example.com:5000" = {
+          auth = {
+            username = "";
+            password = "";
+          };
+          tls = {
+            cert_file = ./path;
+            key_file = ./path;
+            ca_file = ./path;
+          };
+        };
+      };
+      description = lib.mdDoc "[Private registry configuration](https://docs.k3s.io/installation/private-registry).";
```

```suggestion
      description = "[Private registry configuration](https://docs.k3s.io/installation/private-registry).";
```

It's no longer necessary to use `lib.mdDoc`

>           407|   mdDoc = lib.warn "lib.mdDoc will be removed from nixpkgs in 24.11. Option descriptions are now in Markdown by default; you can remove any remaining uses of lib.mdDoc.";